toqito/channel_props/is_extremal.py

```diff
@@ -0,0 +1,64 @@
+"""Determines whether a quantum channel is extremal."""
+
+import numpy as np
+from numpy.linalg import matrix_rank
+
+from toqito.channel_ops.choi_to_kraus import choi_to_kraus
+
+
+def is_extremal(phi: np.ndarray | list[np.ndarray | list[np.ndarray]], tol: float = 1e-9) -> bool:
+    """Determine whether a quantum channel is extremal.
```

Comment:
We will likely want to add a reference to the textbook by John Watrous (The Theory of Quantum Information). Refer to (for instance) this docstring for how to include citations and include a reference section in the docstring:
https://github.com/vprusso/toqito/blob/f93f5168db7dddd053700b2e92a8badea4509cd9/toqito/channel_props/choi_rank.py#L9